### PR TITLE
Add Cosine Similarity

### PR DESCRIPTION
**Pull request by Descartess**
Reference #127

### CODE REVIEW

File: Sources/TensorFlow/Loss.swift [hunk 1]
```diff
@@ -65,6 +65,18 @@ public func hingeLoss<Scalar: TensorFlowFloatingPoint>(
...
+/// Returns the cosine proximity loss between predictions and expectations.
+///
+/// - Parameters:
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+@differentiable(wrt: predicted)
+public func cosineProximityLoss<Scalar: TensorFlowFloatingPoint>(
+    predicted: Tensor<Scalar>, expected: Tensor<Scalar>
+) -> Tensor<Scalar> {
+    return -(expected * predicted).sum() / (sqrt(expected.squared().sum()) * sqrt(predicted.squared().sum()))
```
Comment (rxwei) [inline Sources/TensorFlow/Loss.swift]: Could you please break this line into two so that it fits within 100 columns? The second line should indent by 4 from the first line.

File: Sources/TensorFlow/Loss.swift [hunk 2]
```diff
@@ -65,6 +65,18 @@ public func hingeLoss<Scalar: TensorFlowFloatingPoint>(
     return max(Tensor(1) - expected * predicted, Tensor(0)).mean()
 }
 
+/// Returns the cosine proximity loss between predictions and expectations.
+///
+/// - Parameters:
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+@differentiable(wrt: predicted)
+public func cosineProximityLoss<Scalar: TensorFlowFloatingPoint>(
```
[the remark this answers is not in the document]
Comment (Shashi456) [reply Sources/TensorFlow/Loss.swift]: @rxwei wdyt?

[the remark this answers is not in the document]
Comment (rxwei) [reply Sources/TensorFlow/Loss.swift]: I'd suggest calling this `cosineSimilarity(_:_:)` and make it differentiable w.r.t. both parameters.

Rationale:
1. It is a standard measure of similarity and is the canonical name of this measure (it also has a [Wikipedia page](https://en.wikipedia.org/wiki/Cosine_similarity)).
2. It does not have to be a loss function, so it can be used in other contexts. Hence arguments labels differentiating between predicted and expected do not seem to be useful.
3. Because of (2), we should make it differentiable with respect to both parameters.

[the remark this answers is not in the document]
Comment (rxwei) [reply Sources/TensorFlow/Loss.swift]: I’ve just noticed that the argument labels were not removed as suggested. Could you please start a PR to resolve it when you get a chance? Thanks!

[the remark this answers is not in the document]
Comment (Shashi456) [reply Sources/TensorFlow/Loss.swift]: @rxwei if it's okay I can fix this in #239

[the remark this answers is not in the document]
Comment (rxwei) [reply Sources/TensorFlow/Loss.swift]: Thanks for taking this on! It'd be better if you fix it in a separate CL instead.
